app/components/github_integration/webhooks/issues.py

```diff
@@ -0,0 +1,169 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Literal, Protocol, cast
+
+from app.components.github_integration.webhooks.core import (
+    EmbedContent,
+    Footer,
+    SubhookStore,
+    client,
+    make_subhook_registrar,
+    send_embed,
+)
+
+if TYPE_CHECKING:
+    from githubkit.typing import Missing
+    from githubkit.versions.v2022_11_28.models import (
+        WebhookIssueCommentCreated,
+        WebhookIssuesClosed,
+        WebhookIssuesLocked,
+        WebhookIssuesOpened,
+        WebhookIssuesPinned,
+        WebhookIssuesReopened,
+        WebhookIssuesUnlocked,
+        WebhookIssuesUnpinned,
+    )
+    from monalisten.types import IssueCommentEvent, IssuesEvent
+
+    from app.components.github_integration.emoji import EmojiName
+
+EXPECTED_STATE_REASONS = frozenset({"completed", "not_planned", "duplicate"})
+
+issue_subhooks: SubhookStore[IssuesEvent] = {}
+
+register_issue_subhook = make_subhook_registrar(issue_subhooks)
+
+
+class IssueLike(Protocol):
+    state: Missing[Literal["open", "closed"]]
+    state_reason: Missing[str | None]
+
+
+def get_issue_emoji(issue: IssueLike) -> EmojiName:
+    if issue.state == "open":
+        return "issue_open"
+    suffix = "completed" if issue.state_reason == "completed" else "unplanned"
+    return "issue_closed_" + suffix
```

Comment:
I think you should just make this something like:

```suggestion
    if issue.state_reason == "completed":
        return "issue_closed_completed"
    return "issue_closed_unplanned"
```
as there are only two.